src/common/snippets/src/pass/insert_loops.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <snippets/itt.hpp>
+#include "snippets/pass/insert_loops.hpp"
+#include "snippets/op/loop_helpers.hpp"
+
+#include <ngraph/rt_info.hpp>
+
+ngraph::snippets::pass::InsertLoops::InsertLoops(ov::PartialShape master_shape, size_t vector_size)
+: master_shape(std::move(master_shape)), vector_size(vector_size) {
+}
+
+bool ngraph::snippets::pass::InsertLoops::run_on_model(const std::shared_ptr<ov::Model> &model) {
+    RUN_ON_FUNCTION_SCOPE(InsertLoops);
+    if (master_shape.is_dynamic())
+        throw ngraph_error("InsertLoops doesn't support dynamic shapes yet");
+
+    const auto inner_dim = master_shape.size() - 1;
+    // Note: outer_dim could overflow if master_shape.size() < 2
+    const auto outer_dim = master_shape.size() - 2;
+    const auto inner_WA = master_shape[inner_dim].get_length();
```

Comment:
Agreed)
Let's spell it out then (`inner_work_amount` - everyone uses auto completion anyway), since `size` can be associated with a container size or spatial dimensions.